share/smack/lib/pthread.c

```diff
@@ -0,0 +1,253 @@
+#include "pthread.h"
```

Comment:
Why Mike?
I find it strange to include anything but the respective header file in .c files. At least that is the convention that I am used to. Everything else should be included from the header file, no?